src/librustc_mir/borrow_check/mod.rs

```diff
@@ -2017,13 +2017,13 @@ enum Overlap {
     Disjoint,
 }
 
-impl<'cx, 'gcx, 'tcx> MirBorrowckCtxt<'cx, 'gcx, 'tcx> {
+impl MirBorrowckCtxt<'cx, 'gcx, 'tcx> {
     // FIXME (#16118): function intended to allow the borrow checker
     // to be less precise in its handling of Box while still allowing
     // moves out of a Box. They should be removed when/if we stop
     // treating Box specially (e.g. when/if DerefMove is added...)
 
-    fn base_path<'d>(&self, place: &'d Place<'tcx>) -> &'d Place<'tcx> {
+    fn base_path(&self, place: &'place Place<'tcx>) -> &'place Place<'tcx> {
```

Comment:
any particular reason for the rename here?

Reply:
I think the policy I tried to follow is to avoid single character lifetimes when they're used

Reply:
Huh, ok. This is exactly the case where I would *prefer* a single-character lifetime -- if the point is just to connect a variable and a return value, then a single-character lifetime expresses that (versus trying to tie to some "outer" lifetime).

I think the way I think of it is this:

- longer lifetime names are for lifetimes that cross scopes. 
- single-character lifetimes are for things local to a particular fn
- the names `gcx` and `tcx` are conceptually global; and within a given context, there may be other such names
    - this means in particular that standalone *fn items*  might use longer names to refer to lifetimes from their callers

Not sure if this is the best policy or not but it's the one I was contemplating.

Reply:
I can see how that makes sense; I can switch this case back.